### PR TITLE
ROS-268: support serializable.

### PR DESCRIPTION
**Pull request by benson-basis**

### CODE REVIEW

**Comment by joelb-git** [comment]
@benson-basis 

My build and travis fail with:

```
[INFO] PMD Failure: com.basistech.rosette.dm.SerializableTest:48 Rule:UnusedPrivateField Priority:3 Avoid unused private fields such as 'entityMention'..
[INFO] PMD Failure: com.basistech.rosette.dm.SerializableTest:50 Rule:UnusedPrivateField Priority:3 Avoid unused private fields such as 'resolvedEntity'..
[INFO] PMD Failure: com.basistech.rosette.dm.SerializableTest:66 Rule:UnusedPrivateField Priority:3 Avoid unused private fields such as 'referenceTextOldEntities'.
```

**Comment by benson-basis** [comment]
I don't care about Serializable for the obsolete old entity model, and all these items were that.

File: model/src/main/java/com/basistech/rosette/dm/AnnotatedText.java [hunk 1]
```diff
@@ -47,18 +48,19 @@
  * For Json, and Java, the 'adm-json' module provides the supported serialization.
  */
 @SuppressWarnings("deprecation")
-public class AnnotatedText {
+public class AnnotatedText implements Serializable {
+    private static final long serialVersionUID = 222L;
```
Comment (joelb-git) [inline model/src/main/java/com/basistech/rosette/dm/AnnotatedText.java]: 222?  We'll bump to 223 for next incompatibility?  (I'd probably start at 1...)

File: model/src/main/java/com/basistech/rosette/dm/AnnotatedText.java [hunk 2]
```diff
@@ -47,18 +48,19 @@
...
      * Only one attribute of a type is permitted, thus the concept
      * of a ListAttribute.
      */
     private final Map<String, BaseAttribute> attributes;
     private final Map<String, List<String>> documentMetadata;
-    private boolean compatMentionsProcessed;
-    private ListAttribute<EntityMention> compatMentions;
-    private boolean compatResolvedEntitiesProcessed;
-    private ListAttribute<ResolvedEntity> compatResolvedEntities;
+    private transient boolean compatMentionsProcessed;
```
Comment (joelb-git) [inline model/src/main/java/com/basistech/rosette/dm/AnnotatedText.java]: Transient (and these fields themselves) will go away when we remove the deprecated classes, yes?

File: model/src/main/java/com/basistech/rosette/dm/AnnotatedText.java (hunk 1 above)
Comment (benson-basis) [inline model/src/main/java/com/basistech/rosette/dm/AnnotatedText.java]: I was thinking we'd just put the version number for the release of the ADM when we make a change. But if you think that just starting everything at one would be better, I'll do it.

File: model/src/main/java/com/basistech/rosette/dm/AnnotatedText.java (hunk 2 above)
Comment (benson-basis) [inline model/src/main/java/com/basistech/rosette/dm/AnnotatedText.java]: Yes.

File: model/src/main/java/com/basistech/rosette/dm/Name.java [hunk 3]
```diff
@@ -30,10 +31,11 @@
 /**
  * A name of something in the world.
  * Both Names and {@link com.basistech.rosette.dm.EntityMention} objects can contains the names of things.
```
Comment (joelb-git) [inline model/src/main/java/com/basistech/rosette/dm/Name.java]: EntityMention --> Mention here too?

File: model/src/main/java/com/basistech/rosette/dm/AnnotatedText.java (hunk 1 above)
Comment (joelb-git) [inline model/src/main/java/com/basistech/rosette/dm/AnnotatedText.java]: Oh, 222 is from the version number!  Okay.  Could we document that convention somewhere?

**Pull request closed without merging**